### PR TITLE
Display user email with their comments

### PR DESCRIPTION
**Pull request by mffabrigas**
Add user email to comment entities in datastore and display it alongside the comment text when loaded onto the page

### CODE REVIEW

**Comment by mffabrigas** [comment]
Sorry for the last three unrelated commits, I forgot this PR was still open

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 1]
```diff
@@ -46,10 +49,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<String> demoComments = new ArrayList<String>();
+    List demoComments = new ArrayList();
```
Comment (chaowangzj) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Don't use raw type. Looks like it should be type of `List<List<String>>`

File: portfolio/src/main/java/com/google/sps/data/Marker.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.google.sps.data;
+
+/** Represents a marker on the map. */
+public class Marker {
```
Comment (chaowangzj) [inline portfolio/src/main/java/com/google/sps/data/Marker.java:4]: It is near to the end of the week 5, so you don't have to take my recommendation for this project.

For java value/data class, you can try the AutoValue library to auto generate these boilerplate code:
https://github.com/google/auto/blob/master/value/userguide/index.md

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 3]
```diff
@@ -46,10 +49,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-      demoComments.add(commentContent);
+      List<String> commentElementContent = new ArrayList<String>();
+
+      String commentText = (String) entity.getProperty(CONTENT_TEXT_PROPERTY_NAME);
+      String commentUserEmail = (String) entity.getProperty(USER_EMAIL_TEXT_PROPERTY_NAME);
+
+      commentElementContent.add(commentUserEmail);
+      commentElementContent.add(commentText);
+
+      demoComments.add(commentElementContent);
```
Comment (chaowangzj) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Here is the example why we want to use protocol buffer in real industry. Both server and client needs to know the type, the size of the list and what each element in the list stands for, which is very difficult to maintain and debug. Without using the protocol buffer, you can alternatively put the data into a java class and serialize it into json string.

Again, you don't have to take my recommendation if the time is short, but try to use it in future development.

Comment (mffabrigas) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: The class solution is definitely more convenient like my AuthenticationInfo class, and I will look into using protocol buffers in the capstone project. Thank you!

File: portfolio/src/main/java/com/google/sps/data/Marker.java (hunk 2 above)
Comment (mffabrigas) [reply portfolio/src/main/java/com/google/sps/data/Marker.java:4]: I hadn't heard about AutoValue before, I will look into it for creating value classes in the capstone. Thank you!

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java (hunk 1 above)
Comment (mffabrigas) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: I will fix that, thank you for pointing it out.

Comment (mffabrigas) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Also, why is raw type not good style?

Comment (chaowangzj) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: One philosophy of java programming(or basically all strongly typed language) is that, spotting error at compile time is better than at runtime. Adding type can make sure you don't accidentally add a number into the list and read it out as a list. Such error will be much difficult to debug if it is spotted by client receiving unexpected data from a living server

**Review by chaowangzj — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java (hunk 1 above)
Comment (mffabrigas) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Ah I see, that is good to keep in mind, thank you!

**Pull request merged**